server/src/main/java/com/defold/extender/AsyncBuilder.java

```diff
@@ -92,44 +93,45 @@ public void asyncBuildEngine(MetricsWriter metricsWriter, String platform, Strin
             LOGGER.info("Building engine locally");
 
             // Get SDK
-            final File sdk = defoldSdkService.getSdk(sdkVersion);
-            metricsWriter.measureSdkDownload(sdkVersion);
-
-            extender = new Extender.Builder()
-                        .setPlatform(platform)
-                        .setSdk(sdk)
-                        .setJobDirectory(jobDirectory)
-                        .setUploadDirectory(uploadDirectory)
-                        .setBuildDirectory(buildDirectory)
-                        .setMetricsWriter(metricsWriter)
-                        .build();
-
-            // Resolve Gradle dependencies
-            if (platform.contains("android")) {
-                extender.resolve(gradleService);
-                metricsWriter.measureGradleDownload();
-            }
+            try (DefoldSdk sdk = defoldSdkService.getSdk(sdkVersion)) {
```

Comment:
Wrap `File` with folder path into `DefoldSdk` object and work with it as a resource. When code execution go out from scope - `DefoldSdk.close()` will be called and decrease reference count to sdk.